paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerChoosesArrowScriptEvent.java

```diff
@@ -0,0 +1,97 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.Denizen;
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.*;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.destroystokyo.paper.event.player.PlayerReadyArrowEvent;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class PlayerChoosesArrowScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player chooses arrow
+    //
+    // @Cancellable true
+    //
+    // @Group Player
+    //
+    // @Location true
+    //
+    // @Plugin Paper
+    //
+    // @Triggers when a player chooses an arrow to load a bow/crossbow.
+    //
+    // @Switch arrow:<item> to only process the event when the players chosen arrow matches the input.
+    // @Switch bow:<item> to only process the event when the players bow matches the input.
+    //
+    // @Context
+    // <context.arrow> returns the ItemTag of the arrow that was chosen.
+    // <context.bow> returns the ItemTag of the bow that is about to get loaded.
+    //
+    // @Player Always.
+    //
+    // @Example
+    // # This example prevents to use any arrow but spectral_arrows.
+    // on player chooses arrow:
+    // - if !<context.arrow.advanced_matches[spectral_arrow]>:

```

Comment:
This example is pretty weird. Should just have a switch on the event rather than an `if`

Reply:
also: `prevents using`, not `prevents to use`